packages/clay-css/src/scss/mixins/_badges.scss

```diff
@@ -178,64 +178,106 @@
 		$badge-item-expand: setter(map-get($map, badge-item-expand), ());
 
 		@if ($enabled) {
-			@include clay-css($base);
+			@if (length($base) != 0) {
+				@include clay-css($base);
+			}
 
 			// Empty badges collapse automatically
 
-			&:empty {
-				@include clay-css(map-get($map, empty));
-			}
+			$_empty: map-get($map, empty);
 
-			&[href],
-			&[type] {
-				@include clay-link($href);
+			@if ($_empty) {
+				&:empty {
+					@include clay-css(map-get($map, empty));
+				}
 			}
 
-			a {
-				@include clay-link($link);
+			@if (length($href) != 0) {
+				&[href],
+				&[type] {
+					@include clay-link($href);
+				}
 			}
 
-			.badge-item {
-				@include clay-css($badge-item);
-
+			@if (length($link) != 0) {
 				a {
-					@include clay-link(map-get($badge-item, link));
+					@include clay-link($link);
 				}
+			}
 
-				.btn-unstyled {
-					@include clay-button-variant(
-						map-get($badge-item, btn-unstyled)
-					);
-				}
+			@if (length($badge-item) != 0) {
+				.badge-item {
+					@include clay-css($badge-item);
 
-				.close {
-					@include clay-close(map-get($badge-item, close));
-				}
+					$_link: map-get($badge-item, link);
+
+					@if ($_link) {
+						a {
+							@include clay-link($_link);
+						}
+					}
+
+					$_btn-unstyled: map-get($badge-item, btn-unstyled);
 
-				.lexicon-icon {
-					@include clay-css(map-get($badge-item, lexicon-icon));
+					@if ($_btn-unstyled) {
+						.btn-unstyled {
+							@include clay-button-variant($_btn-unstyled);
+						}
+					}
+
+					$_close: map-get($badge-item, close);
+
+					@if ($_close) {
+						.close {
+							@include clay-close($_close);
+						}
+					}
+
+					$_lexicon-icon: map-get($badge-item, lexicon-icon);
+
+					@if ($_lexicon-icon) {
+						.lexicon-icon {
+							@include clay-css($_lexicon-icon);
+						}
+					}
 				}
 			}
 
-			.badge-item-expand {
-				@include clay-css($badge-item-expand);
+			@if (length($badge-item-expand) != 0) {
+				.badge-item-expand {
+					@include clay-css($badge-item-expand);
 
-				a {
-					@include clay-link(map-get($badge-item-expand, link));
+					$_link: map-get($badge-item-expand, link);
+
+					@if ($_link) {
+						a {
+							@include clay-link($_link);
+						}
+					}
 				}
 			}
 
-			.badge-item-before {
-				@include clay-css(map-get($map, badge-item-before));
+			$_badge-item-before: map-get($map, badge-item-before);
+
+			@if ($_badge-item-before) {
+				.badge-item-before {
+					@include clay-css($_badge-item-before);
+				}
```

Comment:
This is very interesting, in JS you can do all this inline, you save memory and could have the help of inline cache, but interestingly here it could be faster. Maybe this is faster because you pass an empty value to `clay-css`?

Reply:
You have a point. I'll test in the clay-css mixin and others.

Reply:
@matuzalemsteles Sass actually takes a good amount of time processing CSS rule-sets. Processing a simple rule-set:
```
.selector{
    color: red;
}
```
100k times takes 1700ms on my machine. If we wrap it in `@if (false){}` it reduces it to 40ms.

Reply:
@pat270 That's interesting, the bad thing about it is that we need to replicate this rule everywhere.

Reply:
Yeah it's unfortunate, but we only need to apply it to the mixins to get a big boost in build time.